src/components/tools/geometry-tool/geometry-content.tsx

```diff
@@ -1149,18 +1152,21 @@ export class GeometryContentComponent extends BaseComponent<IProps, IState> {
 
     board.on("down", handlePointerDown);
     board.on("up", handlePointerUp);
-    this.handleCreateAxes(board);
   }
 
-  private handleCreateAxes = (board: JXG.Board) => {
+  private handleCreateAxis = (axis: JXG.Line) => {
     const handlePointerDown = (evt: any) => {
       if (!this.props.readOnly) {
         this.handleOpenAxisSettings();
       }
     };
 
+    axis.label && axis.label.on("down", handlePointerDown);
+  }
+
+  private handleCreateAxes = (board: JXG.Board) => {
```

Comment:
I wanted to get rid of this function entirely, but I still call it (in a `setTimeout`) after axis rescaling occurs. For some reason, the axes returned by the axis scaling change are different from the (correct + updated) ones on the board after the `setTimeout` resolves.